ludwig/features/feature_utils.py

```diff
@@ -116,11 +116,11 @@ def get_input_size_with_dependencies(
     """
     input_size_with_dependencies = combiner_output_size
     for feature_name in dependencies:
-        if other_output_features[feature_name].num_fc_layers:
+        if other_output_features[feature_name].decoder_config.num_fc_layers:
             input_size_with_dependencies += other_output_features[feature_name].fc_stack.output_shape[-1]
```

Comment:
previously the fc stack of each output feature was not pert of the decoder, was like a inbetween step between the combiner output and the decoder input. For that reason, parameters like num_fc_stack where assigned to the output feature and not to the decoder. Not sure if this was changed, but here I see something weird: the num_fc_layers is a parameter of the decoder, but in the following line the fc_stack from which the output shape is obtained is not inside the decoder but it's a property of the output feature (like it was before), so maybe this needs to be reconsiliated.

Reply:
Output features still have an fc_stack pre-decoder.  This is my mistake, will fix

Reply:
Whoa, but we did move these params to the decoder though we still instantiate the FC stack in OutputFeatureBase:

Reply:
I think either is fine, if those parameters end up being decoder ones or output feature ones, we should be clear though about which is which and to have consistency ;)

Reply:
After investigating this I made some additional changes:

- Moved fc_stack parameters from decoders back into output feature
- Made threshold a property of BinaryOutputFeatureConfig, not the decoder
- Similarly made clip a property of NumberOutputFeatureConfig
- Output feature's `input_size` should be the output of the combiner (plus any dependencies).  Decoder's `input_size` is the output size of the fc stack.